### ISSUE TITLE
[REVIEW]: Visualizations with statistical details: The 'ggstatsplot' approach

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3167]
**Submitting author:** @IndrajeetPatil (<a href="http://orcid.org/0000-0003-1995-6531">Indrajeet Patil</a>)
**Repository:** <a href="https://github.com/IndrajeetPatil/ggstatsplot" target ="_blank">https://github.com/IndrajeetPatil/ggstatsplot</a>
**Version:** 0.7.2
**Editor:** @csoneson
**Reviewer:** @njtierney, @kevinrue
**Archive:** <a href="https://doi.org/10.5281/zenodo.4681705" target="_blank">10.5281/zenodo.4681705</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/254890248268a43a0365abe1a607939c/status.svg)](https://joss.theoj.org/papers/254890248268a43a0365abe1a607939c)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/254890248268a43a0365abe1a607939c"><img src="https://joss.theoj.org/papers/254890248268a43a0365abe1a607939c/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/254890248268a43a0365abe1a607939c/status.svg)](https://joss.theoj.org/papers/254890248268a43a0365abe1a607939c)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/IndrajeetPatil/ggstatsplot" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@njtierney & @kevinrue, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @csoneson know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/IndrajeetPatil/ggstatsplot">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@IndrajeetPatil) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/IndrajeetPatil/ggstatsplot">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@IndrajeetPatil) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @njtierney, @kevinrue it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by csoneson** [comment]
👋🏼 @IndrajeetPatil, @njtierney, @kevinrue - this is the review thread for the submission. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues directly in the software repository. If you do so, please mention this thread so that a link is created (and I can keep an eye on what is happening). Please also feel free to comment and ask questions in this thread. It is often easier to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

Please feel free to ping me (@csoneson) if you have any questions or concerns. Thanks!

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=4.28 s (58.9 files/s, 24973.2 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
HTML                            68           4624            874          33370
JavaScript                      28           5168           5438          18920
SVG                             26              0              1           7719
R                               60           1030           2680           5930
Markdown                        23           1155              0           5720
Rmd                             21           2070           4358           5056
XML                              2              0              2            606
JSON                             2              1              0            580
CSS                              7            108             58            474
YAML                            12             82             19            444
TeX                              3             29              0            316
-------------------------------------------------------------------------------
SUM:                           252          14267          13430          79135
-------------------------------------------------------------------------------


Statistical information for the repository '7eff902a638cc56c89a03c4a' was
gathered on 2021/04/12.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Indrajeet Patil                 50         54359          15224           99.42
IndrajeetPatil                   5           305             99            0.58

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Indrajeet Patil           39341           72.4          9.4               18.54
```

**Comment by IndrajeetPatil** [comment]
Thanks, @csoneson, will ping you in case of any questions. None for now. 

Looking forward to the comments from the reviewers!

**Comment by kevinrue** [comment]
@csoneson All good from my point of view!

**Comment by IndrajeetPatil** [comment]
@kevinrue Thanks a ton for your helpful feedback! 🙌

**Comment by IndrajeetPatil** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: :point_left:

**Comment by csoneson** [comment]
Thanks a lot @kevinrue!

**Comment by IndrajeetPatil** [comment]
@csoneson Out of curiosity: Is Nicholas Tierny actually assigned to this issue? 
Asking because he is not listed as assignees for this issue.

<img width="173" alt="Screenshot 2021-04-22 at 11 22 43" src="https://user-images.githubusercontent.com/11330453/115690274-2ef49900-a35d-11eb-888b-c1e245d54d60.png">

**Comment by csoneson** [comment]
I think it should still be ok - @njtierney, in case you notice that you are not able to tick the boxes in the checklist, let me know and I'll reinvite you.

**Comment by whedon** [comment]
:wave: @njtierney, please update us on how your review is going (this is an automated reminder).

**Comment by IndrajeetPatil** [comment]
@csoneson Maybe Nicholas Tierny needs to be reinvited?

**Comment by njtierney** [comment]
Hi there, yes can I please be re invited?

**Comment by csoneson** [comment]
@whedon re-invite @njtierney as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@njtierney please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by csoneson** [comment]
@njtierney - could you confirm that you are now able to tick the boxes in the checklist above? Let me know if you have any questions. Thanks!

**Comment by IndrajeetPatil** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3167 with the following error: 
```
 /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:147:in `check_fields': Paper YAML header is missing expected fields: date (RuntimeError)
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:89:in `initialize'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon/processor.rb:38:in `new'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon/processor.rb:38:in `set_paper'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/bin/whedon:58:in `prepare'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'

```

**Comment by IndrajeetPatil** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3167 with the following error: 
```
 /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:204:in `block in parse_authors': Author (Indrajeet Patil) is missing affiliation (RuntimeError)
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:202:in `each'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:202:in `parse_authors'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:93:in `initialize'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon/processor.rb:38:in `new'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon/processor.rb:38:in `set_paper'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/bin/whedon:58:in `prepare'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'

```

**Comment by IndrajeetPatil** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: :point_left:

**Comment by csoneson** [comment]
ping @njtierney - could you let us know how your review is going? Thanks!

> @njtierney - could you confirm that you are now able to tick the boxes in the checklist above? Let me know if you have any questions. Thanks!

**Comment by IndrajeetPatil** [comment]
@csoneson If it turns out he doesn't have time to review right now, is it possible for us to look for an alternative reviewer?
I can suggest a few.

**Comment by csoneson** [comment]
@IndrajeetPatil - let's have some patience, these are challenging times for everyone (in particular, note that JOSS is still operating in [reduced service mode](https://blog.joss.theoj.org/2020/05/reopening-joss) because of the challenges posed by the pandemic).

**Comment by IndrajeetPatil** [comment]
Oh, I see. Thanks for pointing out the reduced service mode, which does state: 

> "In this reduced service mode, we’re asking that reviewers try to complete their reviews in six weeks."

So happy to wait for 3 more weeks and revisit this.

**Comment by IndrajeetPatil** [comment]
@whedon set v0.7.2 as version

**Comment by whedon** [comment]
I'm sorry @IndrajeetPatil, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by csoneson** [comment]
@njtierney - could you let us know how things are going with your review? Thanks!

**Comment by csoneson** [comment]
👋🏻 @njtierney - just checking in on your review 🙂

**Comment by njtierney** [comment]
Hi @csoneson ,

Thanks so much for your patience, 

I've finished up the review now, had a few family/personal things come up which means I had reduced scope for other work.

Linked above are some issues that I noticed reading through the paper and the software. In particular, https://github.com/IndrajeetPatil/ggstatsplot/issues/598 I that it is hard to reproduce the statistical analysis without showing the graphic, I would be interested to discuss this and some solutions to this issue.

**Comment by csoneson** [comment]
Thanks a lot @njtierney!

**Comment by IndrajeetPatil** [comment]
@csoneson Can you please set the version of the package to `0.7.2` in this issue? I can't do it.

There was a new release of the package since the review started, which changes nothing about the paper but better for the record to be up-to-date.

**Comment by csoneson** [comment]
@whedon set v0.7.2 as version

**Comment by whedon** [comment]
OK. v0.7.2 is the version.

**Comment by IndrajeetPatil** [comment]
@njtierney Thanks for your comments!

> In particular, IndrajeetPatil/ggstatsplot#598 I that it is hard to reproduce the statistical analysis without showing the graphic, I would be interested to discuss this and some solutions to this issue.

I have proposed a couple of solutions in the related issue to address this, and I am happy to discuss this further.

**Comment by IndrajeetPatil** [comment]
I see 40/40 checks!!! Thanks, @njtierney! 😄 

@csoneson Does this mean the review process is finished?

**Comment by csoneson** [comment]
Great! Just for the book-keeping, I'd like to confirm with @njtierney and @kevinrue that they are both happy with the current state of the submission and that there are no pending issues that they would still like to see addressed. Then I will take a quick look as well, and then we'll move on with the final steps towards accepting the submission.

**Comment by csoneson** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: :point_left:

**Comment by csoneson** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3758/s13428-015-0664-2 is OK
- 10.21105/joss.01686 is OK
- 10.31234/osf.io/ntbvy is OK
- 10.21105/joss.02815 is OK
- 10.21105/joss.02445 is OK
- 10.21105/joss.01541 is OK
- 10.31234/osf.io/vtq8f is OK
- 10.21105/joss.01412 is OK
- 10.3758/s13423-020-01798-5 is OK
- 10.21105/joss.02306 is OK

MISSING DOIs

- 10.1016/0003-2697(86)90183-1 may be a valid DOI for title: The Elements of Graphing Data

INVALID DOIs

- None
```

**Comment by IndrajeetPatil** [comment]
Thanks, @csoneson. 

I think this note is incorrect though. 

> MISSING DOIs
> -  10.1016/0003-2697(86)90183-1 may be a valid DOI for title: The Elements of Graphing Data

The book in question was published in 1994, way before when `doi`s were introduced (2000). At least, I find no evidence that the suggested `doi` is for this book.

**Comment by csoneson** [comment]
Yes, I agree - that `doi` appears to point to a review of the book.

**Comment by njtierney** [comment]
All good on my end, happy to sign off on the review.

**Comment by kevinrue** [comment]
Same here. I'm happy with the current state of the paper and package.

**Comment by csoneson** [comment]
Excellent, thanks both - @IndrajeetPatil, I'll get back to you shortly.

**Comment by csoneson** [comment]
@IndrajeetPatil - I just sent a PR with some small fixes to the paper and bibliography. Please have a look and merge if you agree. There are a couple of additional things that I think needs fixing in the bibliography: 

- [ ] add all the authors for Doorn et al
- [ ] add ISBNs to all the books for consistency
- [ ] fix the titles of Ludecke et al (2020, 2021) - the ones given in the bibliography are not the same as the papers that the DOIs point to
- [ ] replace the statsExpressions preprint with the link to the JOSS paper (I saw that it was accepted recently)

Once this is done, the next steps are as follows: 

- [ ] Make a tagged release of the software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service
- [ ] Check that the Zenodo archive has the same title and author list as the paper. You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

**Comment by IndrajeetPatil** [comment]
- [ ] add all the authors for Doorn et al
- [ ] add ISBNs to all the books for consistency
- [ ] fix the titles of Ludecke et al (2020, 2021) - the ones given in the bibliography are not the same as the papers that the DOIs point to
- [ ] replace the statsExpressions preprint with the link to the JOSS paper (I saw that it was accepted recently)

Once this is done, the next steps are as follows:

- [ ] Make a tagged release of the software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service
- [ ] Check that the Zenodo archive has the same title and author list as the paper. You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

**Comment by IndrajeetPatil** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3167 with the following error: 
```
 /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `initialize': No such file or directory @ rb_sysopen - 6443d23dcf5a8f6a5d7cad05/paper/paper.yml (Errno::ENOENT)
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `open'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `load_file'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:125:in `load_yaml'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:87:in `initialize'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon/processor.rb:38:in `new'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon/processor.rb:38:in `set_paper'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/bin/whedon:66:in `prepare'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'

```

**Comment by IndrajeetPatil** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3167 with the following error: 
```
 /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `initialize': No such file or directory @ rb_sysopen - c672818537538a8e5b0a041b/paper/paper.yml (Errno::ENOENT)
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `open'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `load_file'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:125:in `load_yaml'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon.rb:87:in `initialize'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon/processor.rb:38:in `new'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/lib/whedon/processor.rb:38:in `set_paper'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/bin/whedon:66:in `prepare'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-92346a0773a4/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'

```

**Comment by IndrajeetPatil** [comment]
@csoneson I am not sure how to solve the compilation error. I don't know which change led to this, so not sure how to solve this 😢

**Comment by csoneson** [comment]
@openjournals/dev - any idea what might be going on here? I'm not sure the issue was introduced by the last round of fixes; right now I'm also not able to compile the version I forked before those changes were made (using the preview service linked in  https://joss.readthedocs.io/en/latest/submitting.html#checking-that-your-paper-compiles).

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: :point_left:

**Comment by arfon** [comment]
> @openjournals/dev - any idea what might be going on here? I'm not sure the issue was introduced by the last round of fixes; right now I'm also not able to compile the version I forked before those changes were made (using the preview service linked in https://joss.readthedocs.io/en/latest/submitting.html#checking-that-your-paper-compiles), which I think should be the same version that whedon compiled a couple of days ago above.

It was the `paper.tex` file that was confusing Whedon. Looks like @IndrajeetPatil removed this and so the compilation now works.

**Comment by IndrajeetPatil** [comment]
Thanks, @arfon! 
Surprised that it's working, because I had tried to compile the paper using http://whedon.theoj.org/ after the commit that removed `.tex` file, but it was still failing.

But I am glad that it's working now! 😄

**Comment by IndrajeetPatil** [comment]
@csoneson All set on my side now! 

I have:
- [ ] checked all items on the checklist you had provided
- [ ] successfully compiled the PDF using `whedon`
- [ ] proof-read it to make sure style and language look okay

Let me know if there is anything else you would like me to change! :)

**Comment by csoneson** [comment]
Excellent, thank you! Only one minor detail I noticed - in the Doorn et al, the first and second name of `Alexander Ly` have been merged, so they're not properly represented. Could you fix that? And since you're anyway modifying that entry, could you also put some braces around `van Doorn` and `van den Bergh`, so that they render nicely together. In the meantime, I'll log the archive and version info.

**Comment by csoneson** [comment]
@whedon set 10.5281/zenodo.4681705 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4681705" target="_blank">10.5281/zenodo.4681705</a> is the archive.

**Comment by csoneson** [comment]
@whedon set 0.7.2 as version

**Comment by whedon** [comment]
OK. 0.7.2 is the version.

**Comment by IndrajeetPatil** [comment]
@csoneson Thanks for pointing this out!

Fixed in https://github.com/IndrajeetPatil/ggstatsplot/commit/8517c2c5a4613f63c99e70dd53c88d030895fd28

**Comment by IndrajeetPatil** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03167/joss.03167/10.21105.joss.03167.pdf) :page_facing_up: :point_left:

**Comment by csoneson** [comment]
Great - I will hand this over now to the associate editor-in-chief on rotation for the last steps!

**Comment by csoneson** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.3758/s13428-015-0664-2 is OK
- 10.21105/joss.01686 is OK
- 10.21105/joss.03236 is OK
- 10.21105/joss.02815 is OK
- 10.21105/joss.02445 is OK
- 10.21105/joss.01541 is OK
- 10.21105/joss.03139 is OK
- 10.21105/joss.01412 is OK
- 10.3758/s13423-020-01798-5 is OK
- 10.21105/joss.02306 is OK

MISSING DOIs

- 10.1016/0003-2697(86)90183-1 may be a valid DOI for title: The Elements of Graphing Data

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2337

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2337, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by csoneson** [comment]
(and I'll just note again that the missing DOI above is a false hit - it's pointing to a review of the book)

**Comment by IndrajeetPatil** [comment]
Yaay!!

Thank you so much @csoneson, @kevinrue, and @njtierney for all your valuable feedback on the manuscript and the software! ❤️

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1397150707359363074) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2338
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03167](https://doi.org/10.21105/joss.03167)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@njtierney, @kevinrue – many thanks for your reviews here and to @csoneson for editing this submission! JOSS relies upon the volunteer efforts of people like you and we simply wouldn't be able to do this without you :sparkles:

@IndrajeetPatil – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03167/status.svg)](https://doi.org/10.21105/joss.03167)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03167">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03167/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03167/status.svg
   :target: https://doi.org/10.21105/joss.03167
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03167/status.svg)](https://doi.org/10.21105/joss.03167)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
